src/common/datadog.rs

```diff
@@ -79,10 +82,26 @@
 /// Gets the base API endpoint to use for any calls to Datadog.
 ///
 /// If `endpoint` is not specified, we fallback to `site`.
-pub(crate) fn get_api_base_endpoint(endpoint: Option<&String>, site: &str) -> String {
-    endpoint
-        .cloned()
-        .unwrap_or_else(|| format!("https://api.{}", site))
+pub(crate) fn get_api_base_endpoint(endpoint: Option<&str>, site: &str) -> String {
+    endpoint.map_or_else(|| format!("https://api.{}", site), compute_api_endpoint)
+}
+
+fn compute_api_endpoint(endpoint: &str) -> String {
+    // This mechanism is derived from the forwarder health check in the Datadog Agent:
+    // https://github.com/DataDog/datadog-agent/blob/cdcf0fc809b9ac1cd6e08057b4971c7dbb8dbe30/comp/forwarder/defaultforwarder/forwarder_health.go#L45-L47
+    // https://github.com/DataDog/datadog-agent/blob/cdcf0fc809b9ac1cd6e08057b4971c7dbb8dbe30/comp/forwarder/defaultforwarder/forwarder_health.go#L188-L190
+    static DOMAIN_REGEX: Lazy<Regex> = Lazy::new(|| {
+        Regex::new(r"(:?[a-z]{2}\d\.)?(datadoghq\.[a-z]+|ddog-gov\.com)/*$")
```

Comment:
dawwwg

Reply:
It would be nice if we could add a local exception. Some of these probably _shouldn't_ get added to the global dictionary just because they happen to show up in a regex or the like, but that seems to be the only solution.

Reply:
I think there is a way to flag exceptions more granularly than global.